src/L2_projection.jl

```diff
@@ -71,21 +67,22 @@ function _mass_qr(::Lagrange{shape, 2}) where {shape <: RefSimplex}
 end
 _mass_qr(ip::VectorizedInterpolation) = _mass_qr(ip.ip)
 
-function Base.show(io::IO, ::MIME"text/plain", proj::L2Projector)
-    println(io, typeof(proj))
-    println(io, "  projection on:           ", length(proj.set), "/", getncells(get_grid(proj.dh)), " cells in grid")
-    println(io, "  function interpolation:  ", proj.func_ip)
-    println(io, "  geometric interpolation: ", proj.geom_ip)
-end
-
-function _assemble_L2_matrix(fe_values, set, dh)
-
-    n = Ferrite.getnbasefunctions(fe_values)
+function _assemble_L2_matrix(dh::DofHandler, qrs_lhs::Vector{<:QuadratureRule})
     M = create_symmetric_sparsity_pattern(dh)
     assembler = start_assemble(M)
+    for (sdh, qr_lhs) in zip(dh.subdofhandlers, qrs_lhs)
```

Comment:
How does this ensure that the pairing is correct? Think about having 2 subdomains and the 2nd subdomain has triangles and quads, so we have 3 sdhs. If I understand this correctly, then this code will fail if we want to do an L2 projection on the second subdomain.

Reply:
Then `qrs_lhs` would include three quadrature rules, one for each subdofhandler?

Reply:
I think the updated syntax solves this better (see updated comment)

Reply:
Okay, probably a different issue than the one resolved in this PR. I was wondering about an L2 projection on true subdomains with mixed grids (and a mixed grid on the full domain).